ui/lib/components/TimeRangeSelector/index.tsx

```diff
@@ -55,7 +55,7 @@ export function toTimeRangeValue(timeRange?: TimeRange): TimeRangeValue {
     return [...t2.value]
   } else {
     const now = dayjs().unix()
-    return [now - t2.value, now]
+    return [now - t2.value, now + 1]
```

Comment:
This seems to be the root cause that you need some wait() in integration tests before. @YiniXu9506 @baurine @shhdgit 

The reason is simple: the current second (e.g. ts=123.56) is not yet finished, so that data happen in this second got truncated when using this second (e.g. ts=123.0) as the end time.

Reply:
But that's wired we change what the recent means. When the user chooses the recent 30s option, we use the recent 31s indeed.

Reply:
No, we are not changing the start time. It is still recent 30s.

Reply:
Oh, I got the wait issue here. Maybe we can use `cy.clock` to specify the global clock.

Reply:
cy.clock will not change what TiDB stores I guess.

Reply:
Yes, but we delayed the timing of the request by one second. emm, it is just like what `wait` do but don't wait.

Reply:
> No, we are not changing the start time. It is still recent 30s.

I got you.. Should recent "30s" imply that there will be 30 "1s" intervals?

Reply:
> > No, we are not changing the start time. It is still recent 30s.
> 
> I got you.. Should recent "30s" imply that there will be 30 "1s" intervals?

It doesn't seem to be necessary...

Reply:
But if we consider the network latency, the user will get the recent 31s data. We have no idea how a common component will be used, so I thought it would be appropriate to guarantee the semantics of the 30 "1s" interval.

Reply:
The real problem is that we are truncating sub-seconds in this component, e.g. 1.466s becomes 1s. This introduces precision lost in the previous "recent X" options.

Additionally, we didn't specify the open-close rule for the range. Is it right closed or right open? We don't know.

If we want to precisely control the "recent" behavior, the above two questions must be resolved first.

Reply:
As a result, as you can see from the test result, the "recent X minutes" does not include the things just happened in this second. This is a bug that doesn't conform to the "recent" semantic totally.

Reply:
> As a result, as you can see from the test result, the "recent X minutes" does not include the things just happened in this second. This is a bug that doesn't conform to the "recent" semantic totally.

I found the "recent" bug too. We should fix the "recent" issue before we do some "recent" related test...